packages/nix/mkDevShells/pf-env-bootstrap-suspend.sh

```diff
@@ -0,0 +1,302 @@
+#!/usr/bin/env bash
+
+# This script suspends by removing the environments EKS cluster nodes and non destructive resources
+
+set -eo pipefail
+export AWS_PAGER=""
+
+####################################################################
+# Step 0: Validation
+####################################################################
+
+REPO_VARIABLES=$(pf-get-repo-variables)
+
+AWS_DIR=$(echo "$REPO_VARIABLES" | jq -r '.aws_dir')
+KUBE_DIR=$(echo "$REPO_VARIABLES" | jq -r '.kube_dir')
+
+KUBE_CONFIG_PATH="$KUBE_DIR/config"
+STATE_FILE="$KUBE_DIR/cluster_suspend_state.json"
+
+if [[ ! -f $KUBE_CONFIG_PATH ]]; then
+  echo "Error: No kubeconfig file found at $KUBE_CONFIG_PATH. Create it with 'pf-update-kube' first!" >&2
+  exit 1
+fi
+
+####################################################################
+# Step 1: Select the cluster
+####################################################################
+
+EKS_CLUSTER_NAME=$(kubectl config get-contexts -o name | fzf --prompt="Select a Kubernetes cluster: ")
+CONTEXT_ARGS="--context=$EKS_CLUSTER_NAME"
+
+####################################################################
+# Step 2: Select the root profile for the cluster
+####################################################################
+
+aws_profile=$(grep '\[profile ' "$AWS_DIR"/config | sed 's/\[profile \(.*\)\]/\1/' | fzf --prompt="Select the AWS Root Profile: ")
+export AWS_PROFILE=$aws_profile
+
+# check that provided profile is the root user
+if [[ ! $(aws sts get-caller-identity --profile "$AWS_PROFILE" --no-cli-page --output json | jq -r '.Arn') =~ "root" ]]; then
+  echo "Error: Provided profile is not the root user. Exiting." >&2
+  exit 1
+fi
+
+####################################################################
+# Step 2: Confirmation
+####################################################################
+
+echo -e "You selected \n" >&2
+echo -e "Cluster: $EKS_CLUSTER_NAME\n" >&2
+echo -e "Profile: $AWS_PROFILE\n" >&2
+echo -e "WARNING: This will reset core cluster utilities. This should only be done as a part of cluster bootstrapping.\n" >&2
+read -rp "Enter name of cluster to confirm: " CONFIRM_CLUSTER
+
+if [ "$EKS_CLUSTER_NAME" != "$CONFIRM_CLUSTER" ]; then
+  echo -e "$CONFIRM_CLUSTER does not match $EKS_CLUSTER_NAME Exiting.\n" >&2
+  exit 1
+fi
+
+####################################################################
+# Step 3: Save the cluster state & update kubeconfig
+####################################################################
```

Comment:
I don't think anything in this step is necessary as long as you don't update the profile which I feel is unnecessary.

Reply:
For the items like previous node count, etc., simply add those as tags to the resource. This has a couple benefits:

- being able to remove this step
- allowing someone to run restore without needing this state file (e.g., running from a different computer)